dired.py

```diff
@@ -1364,6 +1365,44 @@ def vcs_check(self, path, git='git'):
             else:
                 self.vcs_state.update(git=False)
 
+    def call_hg(self, path):
+        hg = self.view.settings().get('hg_path', '')
+        if hg:  # empty string disable hg integration
+            self.hg_thread = threading.Thread(target=self.hg_check, args=(path, hg))
+            self.hg_thread.start()
+        else:
+            self.vcs_state.update(hg=False)
+
+    def hg_check(self, path, hg='hg'):
+        if any(c for c in '~*?[]' if c in hg):
+            match = glob.glob(os.path.expandvars(os.path.expanduser(hg)))
```

Comment:
One more thing.
Vars contain `$` on *nix and `%` on Windows, but `%` is allowed char in filenames, so it would be 

```python 
if any(c for c in '~*?[]$%' if c in hg) and not isfile(hg):
```
or something like this (same for git).